### DISCUSSION TITLE
Alternatives to DbGeometry.PolygonFromText()?

### DISCUSSION THREAD

**Discussion by jdege** [dotMorten/Microsoft.SqlServer.Types#62] [Q&A]
We're trying to move a .NET 4.5.2 project to .NET 5.0, while still using Entity Framework 6, and we've been having issues with the System.Data.Entity.Spatial types.

Some web browsing led us to dotMorten/Microsoft.SqlServer.Types - and that got us only one more step down the path.

Our current problem is that we're making extensive use of the DbGeometry.XxxFromText() methods in our test suites, and it appears that they are not yet implemented:

> [Implement all *FromText methods #17](https://github.com/dotMorten/Microsoft.SqlServer.Types/issues/17)

Since these functions aren't yet implemented, what would be a reasonable workaround? How can I create a DbGeometry, using what is currently implemented?

That is, what can I do instead of this:
> DbGeometry dbGeometry = DbGeometry.PolygonFromText("POLYGON((" +
>     "34.1015625 82.07002819448267, " +
>     "27.0703125 80.23850054635392, " +
>     "35.15625 78.06198918665974, " +
>     "46.40625 78.63000556774836, " +
>     34.1015625 82.07002819448267" + "))", 4326);

**Comment by dotMorten** [comment]
Use Geometry.STGeomFromText

**Reply by jdege** [reply]
Thanks for your quick response.

The code I'm having problems with:
`var geometry = System.Data.Entity.Spatial.DbGeometry.PolygonFromText(knownText, 4326);`

Prior to installing dotMorten.Microsoft.SqlServer.Types, this throws:
> HResult=0x80131509
> Message=Spatial types and functions are not available for this provider because the assembly 'Microsoft.SqlServer.Types' version 10 or higher could not be found.

After installing dotMorten.Microsoft.SqlServer.Types, it throws:
> HResult=0x80004003
> Message=Object reference not set to an instance of an object.

But as for using Geometry.STGeomFromText, there is no STGeomFromText on DbGeometry, there is no Geometry class in namespace System.Data.Entity.Spatial, and none of the Geometry classes I can see seem to be appropriate. (Resharper is suggesting NetTopologySuites.Geometries.Geometry, and that doesn't have an STGeometryFromtext().)

**Reply by jdege** [reply]
As a followup - I do see that there's an STGeomFromText() in SqlGeometry(), and from that I can create a static geometry.

But can I use that against in EntityFramework 6.4 queries?

Or do I have to drop back to ADO?

**Reply by dotMorten** [reply]
I don't know much about EF. @erikej might be able to help

**Reply by ErikEJ** [reply]
Dunno much about EF6 and spatial, sorry.
